Makefile

```diff
@@ -45,10 +45,13 @@ build-e2e: ## Build end-to-end test binary
 	mkdir -p bin
 	$(DOCKER_CMD) go build $(GOGCFLAGS) -o bin/e2e github.com/openshift/machine-api-operator/tests/e2e
 
+test-e2e:
+	go test -v ./test/e2e/...
+
 .PHONY: test
 test: ## Run tests
 	@echo -e "\033[32mTesting...\033[0m"
-	$(DOCKER_CMD) go test ./...
+	$(DOCKER_CMD) go test ./pkg... ./cmd/...
```

Comment:
"./..." would include the new ./test/e2e/... suite

Reply:
> Is there any reason to be more explicit here? I would have expected the previous `./...` to test everything. What breaks without this change?

So this change, I think, explains some questions I had earlier (on slack).

We are using `func Test*` patterns for the e2e tests, which to me seems a bit odd. Presumably we have to call these test patterns out explicitly to prevent running  the e2e tests via `make test`. True?

 If the entry point to the e2e test was `main() + other_code()` then we wouldn't need to do this.

The reason I noticed this was all due to the `*F Framework` in (what appeared to be) unit test code.

Reply:
This is to run disruptive validations for operator assuming a given installer is up and running and a KUBECONFIG is served as an env variable:
Does it update status properly?
Does it reconcile manifests properly?
Does the node link controller link?
Does a machine healthcheck remediate?

Each of those would be a `func Test*` which I find it nice to read so I'd be in favour of going this way, all that Global is doing is avoid to instantiate a client on every function, but I envision all `func Test*` should be self-contained validating a given feature with api calls and waits. Yes, it could be main() + other_code(). The main purpose is to have an easy way for everyone to add new cases to the suite. We could reconsider the entry-point at anytime

Reply:
I'll concede in the interests of pragmatism.

But it violates the principle of least surprise; I saw `func TestFoo()` and as things stand I can't run (unit test) independently.

Reply:
Added the possible alternative here another commit https://github.com/openshift/machine-api-operator/pull/120/commits/7f6822e44fccf0a48c247ff6d9173c99a90ede5b

Reply:
Thanks for making the changes. Looks way better to me.